packages/repeater/src/request-runner/protocols/HttpRequestRunner.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { HttpRequestRunner } from './HttpRequestRunner';
+import { Request, RequestOptions } from '../Request';
+import { RequestRunnerOptions } from '../RequestRunnerOptions';
+import { Protocol } from '../../models';
+import nock from 'nock';
+import 'reflect-metadata';
+import { anything, spy, verify, when } from 'ts-mockito';
+import { Logger, LogLevel } from '@secbox/core';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+
+const createRequest = (options: Partial<RequestOptions> = {}) => {
+  const requestOptions = {
+    url: 'https://foo.bar',
+    method: 'GET',
+    headers: {},
+    ...options
+  };
+  const request = new Request(requestOptions);
+  const spiedRequest = spy(request);
+  when(spiedRequest.method).thenReturn('GET');
+
+  return { requestOptions, request, spiedRequest };
+};
+
+describe('HttpRequestRunner', () => {
+  const setupRunner = (
+    options: RequestRunnerOptions = {},
+    logger?: Logger
+  ): HttpRequestRunner => new HttpRequestRunner(options, logger);
+
+  describe('protocol', () => {
+    const runner = setupRunner();
+    it('should return HTTP', () => expect(runner.protocol).toBe(Protocol.HTTP));
+  });
+
+  describe('run', () => {
+    afterEach(() => {
+      nock.cleanAll();
+    });
+
+    // eslint-disable-next-line jest/expect-expect
+    it('should call setHeaders on the provided request if additional headers were configured globally', async () => {
+      const headers = { testHeader: 'test-header-value' };
+      const runner = setupRunner({ headers });
+      const { request, spiedRequest } = createRequest();
+
+      await runner.run(request);
+
+      verify(spiedRequest.setHeaders(headers)).once();
+    });
+
+    it('should not call setHeaders on the provided request if there were no additional headers configured', async () => {
+      const runner = setupRunner();
+      const { request, spiedRequest } = createRequest();
+
+      await runner.run(request);
+
+      verify(spiedRequest.setHeaders(anything())).never();
+    });
+
+    it('should perform an external http request', async () => {
+      const runner = setupRunner();
+      const { request, requestOptions } = createRequest();
+      nock(requestOptions.url).get('/').reply(200, {});
+
+      const response = await runner.run(request);
+
+      expect(response.statusCode).toBe(200);
+      expect(response.body).toBe('{}');
```

Comment:
As I mentioned above, just verify an entire result object using a partial matching
```suggestion
      expect(response). toMatchObject({ statusCode: 200, body: '{}' });
```